setup.py

```diff
@@ -60,22 +60,35 @@ def run(self):
     name            = name,
     version         = version_ns['__version__'],
     packages        = packages,
-    description     = "Jupyter protocol implementation and client libraries",
+    description     = 'Jupyter protocol implementation and client libraries',
     author          = 'Jupyter Development Team',
     author_email    = 'jupyter@googlegroups.com',
     url             = 'https://jupyter.org',
     license         = 'BSD',
     platforms       = "Linux, Mac OS X, Windows",
     keywords        = ['Interactive', 'Interpreter', 'Shell', 'Web'],
+    project_urls    = {
+        'Documentation': 'https://jupyter-client.readthedocs.io',
+        'Source': 'https://github.com/jupyter/jupyter_client/',
+        'Tracker': 'https://github.com/jupyter/jupyter_client/issues',
+    },
     classifiers     = [
+        'Framework :: Jupyter',
         'Intended Audience :: Developers',
+        'Intended Audience :: Education',
         'Intended Audience :: System Administrators',
         'Intended Audience :: Science/Research',
         'License :: OSI Approved :: BSD License',
+        'Operating System :: MacOS',
+        'Operating System :: Microsoft :: Windows',
+        'Operating System :: POSIX :: Linux',
         'Programming Language :: Python',
         'Programming Language :: Python :: 2.7',
         'Programming Language :: Python :: 3',
         'Programming Language :: Python :: 3.3',
+        'Programming Language :: Python :: 3.4',
+        'Programming Language :: Python :: 3.5',
+        'Programming Language :: Python :: 3.6',
```

Comment:
I feel like I harp on about this, but my preference is to use the `2` and `3` classifiers and leave the more specific ones out. They only get out of date, as these have.

The `python_requires` parameter, which I see you've also added, is a better way of conveying which minor versions it's meant to work with.